### PR TITLE
Start using getOSVersion and getConnectedDevices from appium-ios-device

### PR DESCRIPTION
**Pull request by umutuzgur**
* Start using `appium-ios-device` in the project
* Remove the dependencies to `ideviceinfo` and `idevice_id`

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -17,30 +18,15 @@ const DEFAULT_TIMEOUT_KEY = 'default';
...
-    let udids = _.uniq(_.filter(stdout.split('\n'), Boolean));
-    udid = _.last(udids);
-    if (udids.length > 1) {
-      log.warn(`Multiple devices found: ${udids.join(', ')}`);
-      log.warn(`Choosing '${udid}'. If this is wrong, manually set with 'udid' desired capability`);
-    }
-  } catch (err) {
-    log.errorAndThrow(`Error detecting udid: ${err.message}`);
+  const udids = await utilities.getConnectedDevices();
+  if (udids.length === 0) {
```
Comment (mykola-mokhnach) [inline lib/utils.js]: can also be _.isEmpty(udids)

File: lib/utils.js [hunk 2]
```diff
@@ -17,30 +18,15 @@ const DEFAULT_TIMEOUT_KEY = 'default';
...
+    throw new Error('No device is connected on the host');
   }
-  if (!udid || udid.length <= 2) {
-    throw new Error('Could not detect udid.');
+  if (udids.length > 1) {
+    log.warn(`Multiple devices found: ${udids.join(', ')}`);
+    log.warn(`Choosing '${udid}'. If this is wrong, manually set with 'udid' desired capability`);
```
Comment (mykola-mokhnach) [inline lib/utils.js]: `udid` is not know here

File: lib/utils.js [hunk 3]
```diff
@@ -17,29 +18,14 @@ const DEFAULT_TIMEOUT_KEY = 'default';
...
-      log.warn(`Choosing '${udid}'. If this is wrong, manually set with 'udid' desired capability`);
-    }
-  } catch (err) {
-    log.errorAndThrow(`Error detecting udid: ${err.message}`);
-  }
-  if (!udid || udid.length <= 2) {
-    throw new Error('Could not detect udid.');
+  const udids = await utilities.getConnectedDevices();
+  if (_.isEmpty(udids)) {
+    throw new Error('No device is connected on the host');
```
Comment (mykola-mokhnach) [inline lib/utils.js]: to the host

**Comment by umutuzgur** [comment]
@jlipps Yeah. I think that would be the next thing to do when everything is removed

**Pull request merged**